### ISSUE TITLE
[DHCPv6] Implement DHCP6 client - RFC3315

### ISSUE THREAD

**Issue by danielinux** [tass-belgium/picotcp#231]
[DHCPv6] Implement DHCP6 client - RFC3315

**Comment by S4mw1s3** [comment]
I'd like to know where I can push my code to. I saw that there are branches for AODV and PPP. I assume that I can just create a new branch for this one as well?

**Comment by danielinux** [comment]
yes, you can create your own branch from 'development', and merge later when your unit tests+documentation are in place

**Comment by S4mw1s3** [comment]
FYI, since today I have a working implementation of DHCPv6 but I still need to improve it. Since I'll be starting at a customer on monday, development of this will slow down but I'm not planning to stop. I'll push to a separate branch as soon as I feel comfortable with the implementation. Let me know if you would be interested in an early alpha version.

**Comment by maximevince** [comment]
*like*

**Comment by danielinux** [comment]
+1  for "early alpha". Development branches are for free. :)

**Comment by S4mw1s3** [comment]
I pushed an initial version. This is really still work in progress. See branch dhcp6_client. :)

**Comment by danielinux** [comment]
Hello @S4mw1s3 can you please update your branch merging back the latest development. Also, can you please tell how much is left for this to be merged (i.e. status of development, unit tests, functional tests, documentation...).

Thanks

/d

**Comment by S4mw1s3** [comment]
With merging back you mean I should merge the development branch into the dhcp6_client branch? 

Status of development is still the same. IPv6 DHCP works in a controlled environment. Renewing a dhcp lease when the lease timer is over is also implemented. But most of the corner cases won't work. That is what i mean with a controlled environment. No unit tests are implemented yet. I have not forgotten to continue on this but at the moment its hard to find a few hours where I can do some development. 

The code is not complex since a lot of features are still missing. Therefore if someone wants to help me with development, don't hesitate to ask for help. I'd be glad to share everything I know.

Also is there an easy way how I can test the implementation for rfc compliance? Tahi only runs on certain distributions?

**Comment by danielinux** [comment]
Thanks @S4mw1s3 ! 
@douwedb is picking this up, he will contact you if he needs info.

**Comment by danielinux** [comment]
And yes, TAHI runs on FreeBSD. I have a fully configured VM that I can share if you are interested.

**Comment by S4mw1s3** [comment]
Sure I am. It would be nice to see which exact use cases are already there.

**Comment by danielinux** [comment]
Development moved to https://github.com/tass-belgium/picotcp/tree/dev-dhcp6
